src/mon/OSDMonitor.cc

```diff
@@ -11392,6 +11392,105 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     wait_for_commit(op, new Monitor::C_Command(mon, op, 0, rs,
 						  get_last_committed() + 1));
     return true;
+  } else if (prefix == "osd crush reweight-by-scaling-factor" ||
+             prefix == "osd crush test-reweight-by-scaling-factor") {
+
+    bool really = false;
+    cmd_getval(cmdmap, "yes_i_really_mean_it", really);
+
+    bool skip_metadata_lookup = false;
+    cmd_getval(cmdmap, "skip_metadata_lookup", skip_metadata_lookup);
+
+    bool dry_run = prefix == "osd crush test-reweight-by-scaling-factor";
+
+    if (skip_metadata_lookup)
+      ss << "reweighting all OSDs using scaling factor "
+         << g_conf()->osd_crush_scaling_factor << ":";
+    else
+      ss << "reweighting all OSDs using scaling factor "
+         << g_conf()->osd_crush_scaling_factor
+         << " based on device size metadata:";
```

Comment:
I'm tempted to suggest a ternary operator here, not because if/else is bad, but because ternary operators are cool.